src/rapids_dep_file_generator/cli.py

```diff
@@ -0,0 +1,67 @@
+from .rapids_dep_file_generator import main as dfg
+from ._version import __version__ as version
+from .constants import GeneratorTypes, default_dependency_file_path
+import yaml
+import argparse
+
+
+def generate_file_obj(config_file, file_key, file_type, cuda_version, arch):
+    if not (config_file and file_key and file_type and cuda_version and arch):
+        return {}
+    with open(config_file, "r") as f:
+        parsed_config = yaml.load(f, Loader=yaml.FullLoader)
+    matrix = {"cuda_version": [cuda_version], "arch": [arch]}
+    parsed_config["files"][file_key]["matrix"] = matrix
+    parsed_config["files"][file_key]["generate"] = file_type
+    return {file_key: parsed_config["files"][file_key]}
+
+
+def validate_args(args):
+    mutually_exclusive_arg_keys = ["file_key", "generate", "cuda_version", "arch"]
+    mutually_exclusive_arg_values = []
+    for i in range(len(mutually_exclusive_arg_keys)):
+        mutually_exclusive_arg_values.append(
+            getattr(args, mutually_exclusive_arg_keys[i])
+        )
+    if any(mutually_exclusive_arg_values) and not all(mutually_exclusive_arg_values):
+        raise Exception(
```

Comment:
Strongly avoid raising a plain `Exception`, always pick an error class. Here I'd say `ValueError`.
```suggestion
        raise ValueError(
```